RemoteIDModule/BLE_TX.cpp

```diff
@@ -140,3 +110,105 @@ bool BLE_TX::transmit(ODID_UAS_Data &UAS_data)
 
     return true;
 }
+
+ bool BLE_TX::transmit_legacy(ODID_UAS_Data &UAS_data)
+ {
+	static uint8_t legacy_phase = 0;
+	int legacy_length = 0;
+	// setup ASTM header
+	const uint8_t header[] { 0x1e, 0x16, 0xfa, 0xff, 0x0d, uint8_t(legacy_msg_counter++) };
+	// combine header with payload
+	memset(legacy_payload, 0, sizeof(legacy_payload));
+	memcpy(legacy_payload, header, sizeof(header));
+
+	switch (legacy_phase)
+    {
+		case  0:
+			ODID_Location_encoded location_encoded;
+			memset(&location_encoded, 0, sizeof(location_encoded));
+			if (encodeLocationMessage(&location_encoded, &UAS_data.Location) != ODID_SUCCESS) {
+				break;
+			}
+
+			memcpy(&legacy_payload[sizeof(header)], &location_encoded, sizeof(location_encoded));
+			legacy_length = sizeof(header) + sizeof(location_encoded);
+
+			break;
+
+		case  1:
+			ODID_BasicID_encoded basicid_encoded;
+			memset(&basicid_encoded, 0, sizeof(basicid_encoded));
+			if (encodeBasicIDMessage(&basicid_encoded, &UAS_data.BasicID[0]) != ODID_SUCCESS) {
+				break;
+			}
+
+			memcpy(&legacy_payload[sizeof(header)], &basicid_encoded, sizeof(basicid_encoded));
+			legacy_length = sizeof(header) + sizeof(basicid_encoded);
+
+			break;
+
+		case  2:
+			ODID_SelfID_encoded selfid_encoded;
+			memset(&selfid_encoded, 0, sizeof(selfid_encoded));
+			if (encodeSelfIDMessage(&selfid_encoded, &UAS_data.SelfID) != ODID_SUCCESS) {
+				break;
+			}
+
+			memcpy(&legacy_payload[sizeof(header)], &selfid_encoded, sizeof(selfid_encoded));
+			legacy_length = sizeof(header) + sizeof(selfid_encoded);
+
+			break;
+
+		case  3:
+			ODID_System_encoded system_encoded;
+			memset(&system_encoded, 0, sizeof(system_encoded));
+			if (encodeSystemMessage(&system_encoded, &UAS_data.System) != ODID_SUCCESS) {
+				break;
+			}
+
+			memcpy(&legacy_payload[sizeof(header)], &system_encoded, sizeof(system_encoded));
+			legacy_length = sizeof(header) + sizeof(system_encoded);
+
+			break;
+
+		case  4:
+			ODID_OperatorID_encoded operatorid_encoded;
+			memset(&operatorid_encoded, 0, sizeof(operatorid_encoded));
+			if (encodeOperatorIDMessage(&operatorid_encoded, &UAS_data.OperatorID) != ODID_SUCCESS) {
+				break;
+			}
+
+			memcpy(&legacy_payload[sizeof(header)], &operatorid_encoded, sizeof(operatorid_encoded));
+			legacy_length = sizeof(header) + sizeof(operatorid_encoded);
+
+			break;
+
+		case  5: //broadcast BLE name
```

Comment:
Sometimes things happen outside the standard. It is nice to broadcast also a BLE name, so users can see this name when they scan for nearby Bluetooth devices. I think it would be better to make a different BLE adv service.

Reply:
>  It is nice to broadcast also a BLE name, so users can see this name when they scan for nearby Bluetooth devices

Just for my education. What receiver application are you using to pick up this name? I tried to remove completely the legacy and long range advertisements from the code and only left the code that transmits this name. I did not see anything show up in the nRF Connect app.

Reply:
There are some applications. The first one is that if a user does not have the Remote ID app installed on his phone (like an iPhone), the user can still verify the device is active and broadcasting BLE packets. The second one is that for professional receivers/applications, it is beneficial to know more about the Remote ID device. Which manufacturer/model is currently detected, like a DJI drone, an ArduPilot Remote ID device. Of course the Remote ID already contains a lot of information, but it includes no information about the Remote ID module itself.

Reply:
I meant which iOS/Android application :-). Anyway, now when you changed the code in PR #26 back again, I see what it is you are trying to do. This creates a "Shortened Local Name" data element that is picked up by both the nRF Connect app and is also visible if the phone does a scan for nearby Bluetooth devices (in the Settings menu).

Something similar could have been achieved with the `setScanRspData()`, except that would require the phone to send a scan request to the transmitter, which then responds with the name. At short distances this is not a problem but at Long Range distances this would likely not work and your solution is better.

Reply:
Just FYI: I tested this with a few Android phones and one old iPhone. The Android devices can pick up this name in their Bluetooth scanning part but the iPhone apparently cannot. It is still running iOS 12.5.5 so it is possible that newer iOS versions are able to understand this name element.